user/module/probe_dash.go

```diff
@@ -0,0 +1,311 @@
+	err := b.Module.Init(ctx, logger, conf, ecw)
+	if err != nil {
+		return err
+	}
+	b.conf = conf
+	b.Module.SetChild(b)
+	b.eventMaps = make([]*ebpf.Map, 0, 2)
+	b.eventFuncMaps = make(map[*ebpf.Map]event.IEventStruct)
+	return nil
+}
+
+func (b *MDashProbe) Start() error {
+	if err := b.start(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *MDashProbe) start() error {
+
+	// fetch ebpf assets
+	var bpfFileName = b.geteBPFName("user/bytecode/dash_kern.o")
+	b.logger.Info().Str("bpfFileName", bpfFileName).Msg("BPF bytecode file is matched.")
+	byteBuf, err := assets.Asset(bpfFileName)
+
+	if err != nil {
+		b.logger.Error().Err(err).Strs("bytecode files", assets.AssetNames()).Msg("couldn't find bpf bytecode file")
+		return fmt.Errorf("couldn't find asset %v", err)
+	}
+
+	// setup the managers
+	b.setupManagers()
+
+	// initialize the bootstrap manager
+	if err = b.bpfManager.InitWithOptions(bytes.NewReader(byteBuf), b.bpfManagerOptions); err != nil {
+		return fmt.Errorf("couldn't init manager %v ", err)
+	}
+
+	// start the bootstrap manager
+	if err = b.bpfManager.Start(); err != nil {
+		return fmt.Errorf("couldn't start bootstrap manager %v ", err)
+	}
+
+	// 加载map信息，map对应events decode表。
+	err = b.initDecodeFun()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *MDashProbe) Close() error {
+	if err := b.bpfManager.Stop(manager.CleanAll); err != nil {
+		return fmt.Errorf("couldn't stop manager %v ", err)
+	}
+	return b.Module.Close()
+}
+
+// 通过elf的常量替换方式传递数据
+func (b *MDashProbe) constantEditor() []manager.ConstantEditor {
+	var editor = []manager.ConstantEditor{
+		{
+			Name:  "target_pid",
+			Value: uint64(b.conf.GetPid()),
+			//FailOnMissing: true,
+		},
+		{
+			Name:  "target_uid",
+			Value: uint64(b.conf.GetUid()),
+			//FailOnMissing: true,
+		},
+		{
+			Name:  "target_errno",
+			Value: uint64(b.Module.conf.(*config.DashConfig).ErrNo),
+		},
+	}
+
+	if b.conf.GetPid() <= 0 {
+		b.logger.Info().Msg("target all process.")
+	} else {
+		b.logger.Info().Uint64("target PID", b.conf.GetPid()).Send()
+	}
+
+	if b.conf.GetUid() <= 0 {
+		b.logger.Info().Msg("target all users.")
+	} else {
+		b.logger.Info().Uint64("target UID", b.conf.GetUid()).Send()
+	}
+
+	return editor
+}
+
+func calReadFuncAddressByGDB(elfPath string) (uint64, error) {
+
+	cmd := fmt.Sprintf("%s %s --batch -ex 'printf \"%%p\", %s'", elfPath, "/bin/dash", "read")
+	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
+	if err != nil {
+		return 0, err
+	}
+	hexStr := strings.TrimSpace(string(out))
+	output := strings.TrimPrefix(hexStr, "0x")
+	ui64, err := strconv.ParseUint(output, 16, 64)
+	if err != nil {
+		return 0, err
+	}
+	return ui64, nil
+}
+
+func calReadFuncAddressByObjdump(elfPath string) (uint64, error) {
+	//0000000000005800 <read@plt>:
+	cmd := fmt.Sprintf("%s -d /bin/dash | grep read@plt | head -n 1", elfPath)
+	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
```

Comment:
Executing external commands is not a good practice, can it be achieved using `debug/elf`?  

via: https://github.com/gojue/ecapture/blob/fcddaebc87ff4f8c4991f6f3b5870b300fcded2b/user/config/config_gotls.go#L124-L187